app/assets/javascripts/discourse/app/components/user-preferences/user-passkeys.gjs

```diff
@@ -0,0 +1,253 @@
+        return;
+      }
+
+      this.router.refresh();
+
+      // Prompt to rename key after creating
+      this.dialog.dialog({
+        title: I18n.t("user.passkeys.passkey_successfully_created"),
+        type: "notice",
+        bodyComponent: RenamePasskey,
+        bodyComponentModel: registrationResponse,
+      });
+    } catch (error) {
+      this.errorMessage =
+        error.name === "InvalidStateError"
+          ? I18n.t("user.passkeys.already_added_error")
+          : I18n.t("user.passkeys.not_allowed_error");
+      this.dialog.alert(this.errorMessage);
+    }
+  }
+
+  confirmDelete(id) {
+    schedule("afterRender", () => {
+      this.dialog.deleteConfirm({
+        title: I18n.t("user.passkeys.confirm_delete_passkey"),
+        didConfirm: () => {
+          this.args.model.deletePasskey(id).then(() => {
+            this.router.refresh();
+          });
+        },
+      });
+    });
+  }
+
+  @action
+  async addPasskey() {
+    try {
+      const trustedSession = await this.args.model.trustedSession();
+
+      if (!trustedSession.success) {
+        this.dialog.dialog({
+          title: I18n.t("user.confirm_access.title"),
+          type: "notice",
+          bodyComponent: ConfirmSession,
+          didConfirm: () => this.createPasskey(),
+        });
+      } else {
+        await this.createPasskey();
+      }
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  @action
+  async deletePasskey(id) {
+    try {
+      const trustedSession = await this.args.model.trustedSession();
+
+      if (!trustedSession.success) {
+        this.dialog.dialog({
+          title: I18n.t("user.confirm_access.title"),
+          type: "notice",
+          bodyComponent: ConfirmSession,
+          didConfirm: () => this.confirmDelete(id),
+        });
+      } else {
+        this.confirmDelete(id);
+      }
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  @action
+  renamePasskey(id, name) {
+    this.dialog.dialog({
+      title: I18n.t("user.passkeys.rename_passkey"),
+      type: "notice",
+      bodyComponent: RenamePasskey,
+      bodyComponentModel: { id, name },
+    });
+  }
+
+  <template>
+    <div class="control-group pref-passkeys">
+      <label class="control-label">
+        {{this.title}}
+      </label>
+      <div class="instructions">
+        {{this.instructions}}
+      </div>
+
+      <div class="pref-passkeys__rows">
+        {{#each @model.user_passkeys as |passkey|}}
+          <div class="row">
+            <div class="passkey-left">
+              <div class="row-passkey__name">{{passkey.name}}</div>
+              <div class="row-passkey__created-date">
+                <span class="prefix">
+                  {{this.addedPrefix}}
+                </span>
+                {{this.formatDate
+                  passkey.created_at
+                  format="medium"
+                  leaveAgo="true"
+                }}
```

Comment:
Ideally the date would be a parameter of the translation. Joining those strings together makes it harder to translate.

Reply:
Good call @gschlager, should be addressed now in https://github.com/discourse/discourse/pull/26178